frontend/app/src/App.tsx

```diff
@@ -723,7 +723,8 @@ export class App extends PureComponent<Props, State> {
         delta: (deltaMsg: Delta) =>
           this.handleDeltaMsg(
             deltaMsg,
-            msgProto.metadata as ForwardMsgMetadata
+            msgProto.metadata as ForwardMsgMetadata,
+            msgProto.hash
```

Comment:
Maybe one aspect to double-check is if the hash is always set for all elements. Afaik it is, but maybe something to verify.